pkg/rbac/userData.go

```diff
@@ -108,6 +110,19 @@ func (cache *Cache) GetUserDataCache(ctx context.Context,
 		}
 	}
 
+	// Before checking each namespace and clusterscoped resource, check if user has access to everything
+	userHasAllAccess, err := user.userHasAllAccess(cache, ctx, clientToken)
+	if err != nil {
+		klog.Warning("Encountered error while checking if user has access to everything ", err)
+	} else {
+		if userHasAllAccess {
+			klog.Infof("User %s with uid %s has access to all resources.", userInfo.Username, userInfo.UID)
+			return user, nil
+		}
+		klog.Infof("User %s with uid %s doesn't have access to all resources. Checking individually",
+			userInfo.Username, userInfo.UID)
```

Comment:
Same like above , do we need to log every time ?

Reply:
Updated.